tests/components/renault/test_services.py

```diff
@@ -237,6 +238,101 @@ async def test_service_set_charge_schedule_multi(
     assert mock_call_data[1].thursday.duration == 15
 
 
+async def test_service_set_ac_schedule(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> None:
+    """Test that service invokes renault_api with correct data."""
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    schedules = {"id": 2}
+    data = {
+        ATTR_VEHICLE: get_device_id(hass),
+        ATTR_SCHEDULES: schedules,
+    }
+
+    with (
+        patch(
+            "renault_api.renault_vehicle.RenaultVehicle.get_hvac_settings",
+            return_value=schemas.KamereonVehicleDataResponseSchema.loads(
+                load_fixture("renault/hvac_settings.json")
+            ).get_attributes(schemas.KamereonVehicleHvacSettingsDataSchema),
+        ),
+        patch(
+            "renault_api.renault_vehicle.RenaultVehicle.set_hvac_schedules",
+            return_value=(
+                schemas.KamereonVehicleHvacScheduleActionDataSchema.loads(
+                    load_fixture("renault/action.set_ac_schedules.json")
+                )
+            ),
+        ) as mock_action,
+    ):
+        await hass.services.async_call(
+            DOMAIN, SERVICE_AC_SET_SCHEDULES, service_data=data, blocking=True
+        )
+    assert len(mock_action.mock_calls) == 1
+    mock_call_data: list[ChargeSchedule] = mock_action.mock_calls[0][1][0]
+    assert mock_action.mock_calls[0][1] == (mock_call_data,)
```

Comment:
I think this is wrong (it might be wrong above too) as it's not really testing the values passed.
It's just testing that `A == A`, when it should be testing that `A == expected`

Could we use a snapshot assertion here instead?
```suggestion
    assert mock_call_data == snapshot
```

Reply:
Note: I have opened #125064 for the bad existing tests.
You will probable need to rebase once that is merged.